src/hip_hcc.cpp

```diff
@@ -1405,9 +1406,11 @@ void ihipInit() {
     hsa_status_t err = hsa_iterate_agents(findCpuAgent, &g_cpu_agent);
     if (err != HSA_STATUS_INFO_BREAK) {
         // didn't find a CPU.
-        throw ihipException(hipErrorRuntimeOther);
+        g_initDeviceFound = false;
+        return;
     }
 
+    g_initDeviceFound = true;
```

Comment:
We should probably set this just before returning from ihipInit(). This way we won't have to move this around in case ihipInit() changes and we have another point where an error can be thrown.

Reply:
set g_initDeviceFound to true at the end of ihipInit()